src/SFML/Audio/CMakeLists.txt

```
@@ -67,16 +67,93 @@ if(SFML_OS_IOS)
     set_source_files_properties(${SRCROOT}/Miniaudio.cpp PROPERTIES LANGUAGE OBJCXX)
 endif()
 
-# let CMake know about our additional audio libraries paths (on Android and macOS)
-if(SFML_OS_MACOS)
-    list(APPEND CMAKE_LIBRARY_PATH "${PROJECT_SOURCE_DIR}/extlibs/libs-macos/Frameworks")
-elseif(SFML_OS_ANDROID)
-    list(APPEND CMAKE_INCLUDE_PATH "${PROJECT_SOURCE_DIR}/extlibs/android")
+# find external libraries
+if(SFML_USE_SYSTEM_DEPS)
+    find_package(Vorbis REQUIRED)
+    find_package(FLAC REQUIRED)
+else()
+    # use an immediately invoked function to scope option variables we have to set
+    function(sfml_add_audio_dependencies)
+        include(FetchContent)
+
+        # remember whether we are building SFML as a shared library
+        if(BUILD_SHARED_LIBS)
+            set(SFML_BUILD_SHARED_LIBS ON)
+        endif()
+
+        set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+        set(CMAKE_POLICY_DEFAULT_CMP0048 NEW)
+        set(BUILD_SHARED_LIBS OFF)
+        set(BUILD_CXXLIBS OFF)
+        set(BUILD_PROGRAMS OFF)
+        set(BUILD_EXAMPLES OFF)
+        set(BUILD_TESTING OFF)
+        set(BUILD_UTILS OFF)
+        set(BUILD_DOCS OFF)
+        set(INSTALL_MANPAGES OFF)
+        set(WITH_FORTIFY_SOURCE OFF)
+        set(WITH_STACK_PROTECTOR OFF)
+        set(WITH_AVX OFF) # LLVM/Clang on Windows has issues with AVX2
+
+        FetchContent_Declare(ogg
+            GIT_REPOSITORY https://github.com/xiph/ogg.git
+            GIT_TAG v1.3.5
+            GIT_SHALLOW ON
+            # patch CMAKE_DEBUG_POSTFIX into the Ogg CMake configuration
+            PATCH_COMMAND ${CMAKE_COMMAND} -DOGG_DIR=${FETCHCONTENT_BASE_DIR}/ogg-src -P ${CMAKE_SOURCE_DIR}/tools/ogg/PatchOgg.cmake)
```

```suggestion
            PATCH_COMMAND ${CMAKE_COMMAND} -DOGG_DIR=${FETCHCONTENT_BASE_DIR}/ogg-src -P ${CMAKE_CURRENT_LIST_DIR}/../../../tools/ogg/PatchOgg.cmake)
```

I've had to change these lines here, otherwise projects adding SFML itself via `FetchContent` would not find the `.cmake` file. Not sure what the idiomatic solution is, but this worked...

---

`PROJECT_SOURCE_DIR` is the idiomatic solution.